src/react-native-animated-charts/src/charts/linear/ChartPathProvider.tsx

```diff
@@ -93,11 +144,15 @@ function createPath({ data, width, height, yRange }: CallbackType): PathData {
     });
   }
 
+  // Force linear curve for flat or nearly flat data to prevent spikes
+  const isFlatData = detectFlatData(data.points) || detectNearlyFlatData(data.points);
```

Comment:
Isn't this redundant? Since it's already called in `getScales`. I'd probably just have `getScales` return `isFlatData`.

And you could move the `isNearlyFlatData` logic to within `detectFlatData` and consolidate to a single boolean, since flat/nearly flat are treated in exactly the same way.

Reply:
Ah sorry, I see a linear curve is used only for flat data. So if it's the case that nearly flat data shouldn't use a linear curve, you could keep the two booleans but still remove the call redundancy.

Reply:
i actually think it makes sense to use a Linear curve for nearly flat data as well. Wdyt?